ballerina/commons.bal

```diff
@@ -16,6 +16,47 @@
 
 import ballerina/time;
 
+# Represents the task service.
+public type Service distinct service object {
+  isolated function onTrigger() returns error?;
+};
+
+# Worker count for the global scheduler
+public configurable int globalSchedulerWorkerCount = 5;
+
+# Waiting time for the global scheduler
+public configurable time:Seconds globalSchedulerWaitingTime = 5;
+
+# Listener configuration.
+# 
+# + schedule - The schedule configuration for the listener
+public type ListenerConfiguration record {
+  OneTimeConfiguration|RecurringConfiguration schedule;
+};
+
+# Recurring schedule configuration.
+# 
+# + interval - The duration of the trigger (in seconds), which is used to run the job frequently
+# + maxCount - The maximum number of trigger counts
+# + startTime - The trigger start time in Ballerina `time:Civil`. If it is not provided, a trigger will
+#               start immediately
+# + endTime - The trigger end time in Ballerina `time:Civil`
+# + taskPolicy - The policy, which is used to handle the error and will be waiting during the trigger time
+public type RecurringConfiguration record {|
+  decimal interval;
+  int maxCount = -1;
+  time:Civil startTime?;
+  time:Civil endTime?;
+  TaskPolicy taskPolicy = {};
```

Comment:
I assume `{}` value will set the default `TaskPolicy`. Can't we have a constant for that and set it here, rathe than specifying `{}` ?

Reply:
The `TaskPolicy` has default values for all of its record. There when specifying as `{}`, it will get the default values.

```ballerina
public type TaskPolicy record {|
   ErrorPolicy errorPolicy = LOG_AND_TERMINATE;
   WaitingPolicy waitingPolicy = WAIT;
|};
```